storage/spanstore/reader/reader.go

```diff
@@ -414,10 +420,11 @@ func (s *SpanReader) spansFromPartition(ctx context.Context, traceID model.Trace
 		if err != nil {
 			return err
 		}
-		defer res.Close()
```

Comment:
Could you please clarify the purpose of this diff?

Reply:
some linters (errcheck for example) required check error result on res.Close()